aten/src/ATen/native/group_norm.cpp

```diff
@@ -133,5 +133,29 @@ Tensor group_norm(
 DEFINE_DISPATCH(GroupNormKernel);
 DEFINE_DISPATCH(GroupNormBackwardKernel);
 
+std::tuple<at::Tensor, at::Tensor, at::Tensor> math_group_norm(
+    const at::Tensor& input, const at::Tensor& weight,
+    const at::Tensor& bias, int64_t N, int64_t C, int64_t HxW,
+    int64_t group, double eps) {
+  auto input_shape = input.sizes();
+  at::Tensor input_reshaped = input.view({1, N * group, N ? -1 : 1});
+  auto outputs = at::native_batch_norm(
+      input_reshaped, /*weight=*/{}, /*bias=*/{}, /*running_mean=*/{},
+      /*running_var=*/{}, /*training=*/true, /*momentum=*/0, eps);
+  at::Tensor out = std::get<0>(outputs);
+  out = out.view(input_shape);
+  std::vector<int64_t> affine_param_shape(input.dim(), 1);
+  affine_param_shape[1] = C;
+  if (weight.defined() && bias.defined()) {
+    out = bias.view(affine_param_shape).addcmul(out, weight.view(affine_param_shape), 1);
+  } else if (weight.defined()) {
+    out = out.mul(weight.view(affine_param_shape));
+  } else if (bias.defined()) {
+    out = out.add(bias.view(affine_param_shape));
+  }
+  at::Tensor mean = std::get<1>(outputs).view({N, group});
+  at::Tensor rstd = std::get<2>(outputs).view({N, group});
+  return std::make_tuple(out, mean, rstd);
```

Comment:
I'm going to trust that this was lifted directly from XLA and has already been code reviewed.